src/gluonts/model/deepar/_network.py

```diff
@@ -266,19 +266,27 @@ def unroll_encoder(
         # (batch_size, sub_seq_len, input_dim)
         inputs = F.concat(input_lags, time_feat, repeated_static_feat, dim=-1)
 
+        if F is mx.ndarray:
+            ctx = (
+                inputs.context
+                if isinstance(inputs, mx.gluon.tensor_types)
+                else inputs[0].context
+            )
+            with ctx:
+                begin_state = self.rnn.begin_state(
+                    func=F.zeros, dtype=self.dtype, batch_size=inputs.shape[0]
+                )
+        else:
+            begin_state = self.rnn.begin_state(
+                func=F.zeros, dtype=self.dtype, batch_size=0
+            )
```

Comment:
It looks to me like this context logic should not be in the network code. Maybe it would be better to control the context once and for all from the outside somehow? Otherwise I can see how the code of every model could be filled with similar context logic, and lose readability. What do you think?

Reply:
Agree and in fact this has been repeated. Though I was expecting to have this in upstream mxnet to not hack this into.